sdk/src/stake_weighted_timestamp.rs

```diff
@@ -85,15 +84,16 @@ where
             // estimate offset since the start of the epoch is higher than
             // `MAX_ALLOWABLE_DRIFT_PERCENTAGE_SLOW`
             estimate = epoch_start_timestamp
-                + poh_estimate_offset.as_secs() as i64
-                + max_allowable_drift_slow.as_secs() as i64;
+                .saturating_add(poh_estimate_offset.as_secs() as i64)
+                .saturating_add(max_allowable_drift_slow.as_secs() as i64);
         } else if estimate_offset < poh_estimate_offset
             && poh_estimate_offset - estimate_offset > max_allowable_drift_fast
```

Comment:
Oh 😢 , I was letting clippy lead the way. Guess we'll have to go line-by-line

Reply:
Ah, Duration saturating ops are nightly-only. I'm inclined to leave as-is for now, rather than add a bunch of `checked_math(..).unwrap_or(MAX/MIN)` that makes it even harder to read

Reply:
I'm game to leave them for now.

WDYT about a helper trait that matches the nightly saturating API, but does the `checked_*().unwrap_or(MIN/MAX)` so that we have a quick way to find all of the instances and a zero-effort migration once the core API is available

Reply:
Wfm. Separate PR okay?

Reply:
:100:

Reply:
@t-nelson: because the Duration methods are already live in nightly, looks like we'd either have to use different method names, or fully qualified syntax to call the trait methods. This somewhat defeats the purpose, since it won't be a drop-in replacement when the core API is available. Still worth it?

Reply:
Hmm, I guess nightly clippy hits us? Pity... It feels like a bit of a waste to not flag these while we're doing the initial pass. Maybe use the API names with a `sol_` prefix for easy `sed`ing later?